integration/mediation-tests/tests-mediator-2/src/test/java/org/wso2/carbon/esb/mediator/test/script/SetRemovePropertiesWithNashornJsTestCase.java

```diff
@@ -0,0 +1,135 @@
+import org.wso2.esb.integration.common.utils.ESBIntegrationTest;
+import static org.testng.Assert.assertNotNull;
+
+/**
+ * This test case verifies that, setting and removing properties with NashornJs happens correctly.
+ */
+public class SetRemovePropertiesWithNashornJsTestCase extends ESBIntegrationTest {
+
+    private LogViewerClient logViewerClient;
+
+    @BeforeClass(alwaysRun = true)
+    public void setEnvironment() throws Exception {
+        super.init();
+        logViewerClient = new LogViewerClient(contextUrls.getBackEndUrl(), getSessionCookie());
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void destroy() throws Exception {
+        super.cleanup();
+    }
+
+    @Test(groups = "wso2.esb", description = "Set a property with axis2 scope in script mediator")
+    public void testSetPropertyWithAxis2ScopeInScript() throws Exception {
+
+        boolean setPropertyInLog = false;
+        boolean removePropertyInLog = false;
+        OMElement response = axis2Client.sendCustomQuoteRequest(getProxyServiceURLHttp
+                ("setRemovePropertiesWithNashornJsTestProxy"), null, "inlineTest");
+        LogEvent[] logs = logViewerClient.getAllRemoteSystemLogs();
+        assertNotNull(response, "Response message null");
+
+        for (LogEvent logEvent : logs) {
+            String message = logEvent.getMessage();
+            if (message.contains("Axis2_Property = AXIS2_PROPERTY")) {
+                setPropertyInLog = true;
+                break;
+            }
+        }
+
+        Assert.assertTrue(setPropertyInLog, " The property with axis2 scope is not set ");
+
+        for (LogEvent logEvent : logs) {
+            String message = logEvent.getMessage();
+            if (message.contains("Axis2_Property_After_Remove = null")) {
+                removePropertyInLog = true;
+                break;
+            }
+        }
+
+        Assert.assertTrue(removePropertyInLog, " The property with axis2 scope is not removed ");
+    }
+
+    @Test(groups = "wso2.esb", description = "Set a property with transport scope in script mediator")
+    public void testSetPropertyWithTransportScopeInScript() throws Exception {
+
+        boolean setPropertyInLog = false;
+        boolean removePropertyInLog = false;
+        OMElement response = axis2Client.sendCustomQuoteRequest(getProxyServiceURLHttp
+                ("setRemovePropertiesWithNashornJsTestProxy"), null, "inlineTest");
+        LogEvent[] logs = logViewerClient.getAllRemoteSystemLogs();
+        assertNotNull(response, "Response message null");
+        for (LogEvent logEvent : logs) {
+            String message = logEvent.getMessage();
+            if (message.contains("Transport_Property = TRANSPORT_PROPERTY")) {
+                setPropertyInLog = true;
+                break;
+            }
+        }
+        Assert.assertTrue(setPropertyInLog, " The property with transport scope is not set ");
+
+        for (LogEvent logEvent : logs) {
```

Comment:
Since this code segment seems to be common, can we create a separate method for this?